builtin/logical/pki/path_acme_test.go

```diff
@@ -396,6 +397,29 @@ func TestAcmeAccountsCrossingDirectoryPath(t *testing.T) {
 	// swallows the error we are sending back to a no account error
 }
 
+// TestAcmeDisabledWithEnvVar verifies if VAULT_DISABLE_PUBLIC_ACME is set that we completely
+// disable the ACME service
+func TestAcmeDisabledWithEnvVar(t *testing.T) {
+	os.Setenv("VAULT_DISABLE_PUBLIC_ACME", "true")
+	defer os.Unsetenv("VAULT_DISABLE_PUBLIC_ACME")
```

Comment:
I recently learned about https://beta.pkg.go.dev/testing@master#T.Setenv, in case you want to use that. As the doc notes, it still doesn't allow for parallel execution, but should take care of unsetting for you.